packages/agent/src/agent-main.ts

```diff
@@ -46,10 +47,32 @@ export async function agentMain(argv: string[]): Promise<App> {
   const { baseUrl, clientId, clientSecret, agentId } = args;
 
   const medplum = new MedplumClient({ baseUrl, clientId });
-  await medplum.startClientLogin(clientId, clientSecret);
+
+  let loggedIn = false;
+  while (!loggedIn) {
+    try {
+      await medplum.startClientLogin(clientId, clientSecret);
+      loggedIn = true;
+    } catch (err) {
+      console.error('Failed to login', { err: normalizeErrorString(err) });
+      console.log('Retrying login in 10 seconds...');
+      await sleep(RETRY_WAIT_DURATION_MS);
+    }
+  }
 
   const app = new App(medplum, agentId, parseLogLevel(args.logLevel ?? 'INFO'));
-  await app.start();
+
+  let started = false;
+  while (!started) {
+    try {
+      await app.start();
```

Comment:
Do we know what the conditions are that can fail at this point?

Reply:
Yeah maybe it's actually not a good idea to catch here because there could be real errors we want to throw and stop the service. With restarts it should be more robust to that now anyways

Reply:
In rare cases fetch could fail when attempting to fetch the config from Medplum server, but that should be pretty rare since we basically make sure we are connected in two spots before we get to that line (the call to `App.startWebSocket` just before and also the `startClientLogin` loop we have